lightning-invoice/src/utils.rs

```diff
@@ -381,31 +399,43 @@ where
 /// channel with a higher or equal inbound capacity than `min_inbound_capacity_msat` exists
 /// * If any public channel exists, the returned `RouteHint`s will be empty, and the sender will
 /// need to find the path by looking at the public channels instead
-fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>) -> Vec<RouteHint>{
+fn filter_channels<L: Deref>(
+	channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>, logger: &L
+) -> Vec<RouteHint> where L::Target: Logger {
 	let mut filtered_channels: HashMap<PublicKey, &ChannelDetails> = HashMap::new();
 	let min_inbound_capacity = min_inbound_capacity_msat.unwrap_or(0);
 	let mut min_capacity_channel_exists = false;
 
+	log_debug!(logger, "Considering {} channels for invoice route hints", channels.len());
 	for channel in channels.iter() {
 		if channel.get_inbound_payment_scid().is_none() || channel.counterparty.forwarding_info.is_none() {
+			log_debug!(logger, "Ignoring channel {} for invoice route hints", log_bytes!(channel.channel_id));
 			continue;
 		}
 
 		if channel.is_public {
 			// If any public channel exists, return no hints and let the sender
 			// look at the public channels instead.
+			log_debug!(logger, "Not including invoice route hints on account of public channel {}",
+				log_bytes!(channel.channel_id));
 			return vec![]
 		}
 
 		if channel.inbound_capacity_msat >= min_inbound_capacity {
-			min_capacity_channel_exists = true;
+			if !min_capacity_channel_exists {
+				log_debug!(logger, "Channel with enough inbound capacity exists for invoice route hints");
```

Comment:
nit: should we print channel id here too like in the diffs above?

Reply:
Prefer to leave it out as it (and any others meeting the criteria) will be printed later after filtering.